src/main/scala/it/pagopa/interop/purposeprocess/common/readmodel/ReadModelQueries.scala

```diff
@@ -119,7 +119,11 @@ object ReadModelQueries {
     // Note: the filter works on the assumption that if a version in Draft exists, it is the only version in the Purpose
     val versionsFilter = Filters.or(
       Filters
-        .and(Filters.eq("data.consumerId", requesterId.toString), Filters.eq("data.versions.state", Draft.toString)),
+        .and(
+          Filters.eq("data.consumerId", requesterId.toString),
+          Filters.ne("eservice.data.producerId", requesterId.toString),
```

Comment:
what happens if I'm both the producer and the consumer, and I'm doing this request as the consumer to see my draft purposes?

Reply:
I asked to Carmine some consumer id to test the different behaviour with or without this predicate.

consumerId: 5ec5dd81-ff71-4af8-974b-4190eb8347bf (consumer is also producer) without this predicate 7 results, with this predicate 5 results seems that exclude services for which it also is the producer.

consumerId: 05c7ad1f-84ca-41f5-932a-9cb1e0187e75 (consumer is not producer ) with or without the predicate always 7 results.

It seems to work!

Reply:
When the user is on the producer view is perfect, but what if the first consumer (`5ec5dd81-ff71-4af8-974b-4190eb8347bf`) access the list of its own purposes, which contains a Draft version for an EService that it owns?
Isn't the Draft purpose hidden in this case?

Reply:
IMHO 1 it's a no-go. Both 2 and 3 will affect FE devs user experience, maybe it's worth asking them?

Reply:
Yes, Alessio decided to change splitting the BFF endpoint (the former for producer and the latter for the consumer) and passing a further parameter excludeDraft to achieve the aim of excluding the DRAFT purposes from the query